### PR TITLE
Add ability to compile scss in joomla! template manager

### PR DESCRIPTION
**Pull request by dgrammatiko**
Uses SCSS compiler written in PHP from http://leafo.net/scssphp
MIT & GPL V2

### CODE REVIEW

**Comment by nternetinspired** [comment]
Hi @dgt41 

I really appreciate the effort you've gone to, and the time taken to prepare this PR for me, but this isn't something I'd like to include in Oneweb, for a couple of reasons. 
1. I'm very much against compiling on the server, for a number of reasons, and it's something I disable in other templates as soon as I see it (http://internet-inspired.com/wrote/tutorials/disabling-gantrys-less-compiler/). In fact, I'm even against the ability to edit templates at all from within the template manager, I'd rather that wasn't possible.
2. As soon as I get a moment I'll be updating the Oneweb template with the latest version of my Sass  framework, which is now independent from the template (https://github.com/nternetinspired/oneweb-sass) and depends on Sass 3.3+ which this compiler does not support.

Sorry. Thanks again for your efforts though, they are still appreciated.

File: compile.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
+/* =====================================================================
```
Comment (nternetinspired) [inline compile.php:2]: Just an FYI really, but as this is a new addition then this attribution block should reference both your and the author of scssphp's contributions, as none of this was written by me :)

**Comment by dgrammatiko** [comment]
Hi @nternetinspired,
I didn't know that you were against compiling on server. Reading your post I can see why and I agree with you on most points. Codekit is the king, you're right. But this is somehow different, the css is rendered only if an administrator actually hits the button "compile", I mean that the code will not run in every save on the template manager (as it is with so many frameworks), which I also find it very annoying. Therefore I will close this PR.

Thanks for all the hard work you and that you sharing it with us...

**Comment by nternetinspired** [comment]
I agree with you there, it's great that it doesn't run all the time, but I also think such tools are too dangerous to be used from the admin area, as there is no undo option.

When you edit scss/less/css in a code editor you can undo your changes, on a live site you are screwed. With so many interdependencies a small change can destroy a site quite easily, something I've seen many times when novices edit php files on their server.

If I had my way it wouldn't be possible to edit any file directly from within Joomla, because I think the CMS should be the safety net between users and the filesystem.

You should keep your own fork alive though and when someone asks me for compiling on the server I can point them to your fork :) In fact I may even add that to the readme.

**Comment by dgrammatiko** [comment]
That's an easy one :) Just mention that it's not SASS 3.3 compatible. Maybe I'll make a PR on Joomla! for a better compiler, but then again there is a PR for Composer, so I guess that will not be necessary...

**Pull request closed without merging**